### ISSUE TITLE
why the truck's heatmap does not show anything？

### ISSUE THREAD

**Issue by mengjingyouling** [pooya-mohammadi/yolov5-gradcam#3]
why the truck's heatmap does not show anything？

**Comment by pooya-mohammadi** [comment]
Hi @mengjingyouling 
Unfortunately, I don't have the answer as well. I checked the code several times and tested it on several images, but mysteriously it does not return a visual heatmap for the truck!

**Comment by mengjingyouling** [comment]
@pooya-mohammadi Tried some other truck images, which can display the heatmap normally. The truck occupies a larger size in these pictures. Is it related to the size of the object that needs to be identified?

**Comment by mengjingyouling** [comment]
@pooya-mohammadi I did some experiments to further verify my point . The car / truck in the upper right corner of the dog.jpg image is cropped out and used as a new image dog_new  alone. Then to run the code to do heatmap visualization. As a result, you can see that the visual code is running normally.
But, it seems that the corresponding position of the heatmap is not perfect.

**Comment by pooya-mohammadi** [comment]
@mengjingyouling 
Tnx for your interest in the project. Let me check your hypothesis. I will let you know once it's done.

**Comment by mengjingyouling** [comment]
@pooya-mohammadi Is it the function  Box.fill_ outer_box  caused the problem?

Since the original image will be scaled to 640, it leads to inaccurate position correspondence.

**Comment by mengjingyouling** [comment]
@pooya-mohammadi 
I did some experiments. Useing model_23 layer parameter，the truck in dog.jpg show no heat map. But in shallow layers, such as model_ 3 there are some heat maps.
Can it be understood that the recognition of trucks is determined by shallow features?THX.

**Comment by pooya-mohammadi** [comment]
@mengjingyouling 
I don't think so because in the images you emailed me the truck had a clear heatmap like the others which means that layer is capable of generating visible gradients and heatmaps for trucks
